### PR TITLE
locust metric with experiment generation

### PR DESCRIPTION
**Pull request by tperol**
New locust metrics example using experiment generation and our trial pod for Locust.

### CODE REVIEW

File: locust-metrics/README.md [hunk 1]
```diff
@@ -1,3 +1,47 @@
...
+  latency: p95
+```
+
+Generate the `experiment.yaml` file using
+```
+redskyctl generate experiment -f app.yaml > experiment.yaml
+```
+Generate the RBAC permission that allow the controller to patch the deployment:
+```
+redskyctl generate rbac -f experiment.yaml
```
Comment (jgustie) [inline locust-metrics/README.md:42]: Wouldn't this need to go out to file as well?

File: locust-metrics/locustfile.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import os
+
+from uuid import uuid4
+import random
+from locust import HttpUser, task, between
+
+
+CAT_FRACTION = os.getenv("CAT_FRACTION", 0.75)
```
Comment (jgustie) [inline locust-metrics/locustfile.py:8]: How did cat fraction get this high? What is going on here?

File: locust-metrics/README.md (hunk 1 above)
Comment (tperol) [reply locust-metrics/README.md:42]: forgot to apply the file generated, thanks for catching that.

File: locust-metrics/locustfile.py (hunk 2 above)
Comment (tperol) [reply locust-metrics/locustfile.py:8]: great point, let me change that, this is ridiculous: 🐶 > 🐱

File: locust-metrics/app.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+# directories, URLs, Git repos, etc.
+resources:
+- https://github.com/thestormforge/examples/voting-webapp/application
+
+# This section configures the objectives of optimization
+# We want to minimize both cost and latency
+objectives:
+- name: cost-gcp # Also "cost", "cost-aws", "cpu", ...
+- name: p95-latency
+  latency: p95
```
Comment (jgustie) [inline locust-metrics/app.yaml:17]: Technically you don't need both of these: we [strip the word "latency"](https://github.com/thestormforge/optimize-controller/blob/31298dbee2ac2a1fba810e619832b0935dd123df/api/apps/v1alpha1/application_defaults.go#L139) from the name when trying to [decide what the default value](https://github.com/thestormforge/optimize-controller/blob/31298dbee2ac2a1fba810e619832b0935dd123df/api/apps/v1alpha1/latency.go#L30) should be.

**Review by jgustie — APPROVED** [review]

File: locust-metrics/app.yaml (hunk 3 above)
Comment (tperol) [reply locust-metrics/app.yaml:17]: ok let me remove it then.

File: locust-metrics/README.md [hunk 4]
```diff
@@ -1,3 +1,47 @@
...
+- the experiment generator (controller v1.9.0+ required)
+- the metrics built into Locust
+
+This is useful when tuning an application that does not have metrics yet.
+
+## Deploy the voting app
+
+Run
+```
+kustomize build . | kubectl apply -f -
```
Comment (stephanie-rifai) [inline locust-metrics/README.md:13]: should we change this to the github repo command?
```
kustomize build github.com/thestormforge/examples/voting-webapp/application | \
  kubectl apply -f -
```

Comment (stephanie-rifai) [reply locust-metrics/README.md:13]: okay the problem was that I tried to go through this example without cloning the repo, if we assume people will clone it first then ignore me

File: locust-metrics/locustfile.py (hunk 2 above)
Comment (jdplatt) [reply locust-metrics/locustfile.py]: I think I deliberately snuck it in when I had a big PR

File: locust-metrics/README.md [hunk 5]
```diff
@@ -1,3 +1,47 @@
...
+    spawnRate: 20
+    locustfile: locustfile.py
+```
+
+We optimize for cost and p95 latency as provided by Locust:
+```
+objectives:
+- name: cost-gcp # Also "cost", "cost-aws", "cpu", ...
+- name: p95-latency
+  latency: p95
```
Comment (jgustie) [inline locust-metrics/README.md:33]: It got updated in the file but not in the README, do we need to duplicate the full file contents in the README?

**Review by jgustie — DISMISSED** [review]

File: locust-metrics/README.md (hunk 5 above)
Comment (tperol) [reply locust-metrics/README.md]: fixed

**Review by jgustie — APPROVED** [review]

**Pull request merged**